src/OrchardCore.Modules/OrchardCore.Users/Views/UserFields.Edit.cshtml

```diff
@@ -40,20 +43,81 @@
         </div>
 
         <div class="mt-2">
-            <div class="btn btn-warning btn-sm" id="generatePassword">@T["Generate password"]</div>
+            <div class="btn btn-warning btn-sm" id="generatePassword" data-password-length="@Math.Max(IdentityOptions.Value.Password.RequiredLength, IdentityOptions.Value.Password.RequiredUniqueChars)">@T["Generate password"]</div>
```

Comment:
What if we used `IdentityOptions.Value.Password.RequiredLength` here?

Reply:
Let say if the user set ` IdentityOptions.Value.Password.RequiredUniqueChars` to 8 for some strange reason and `IdentityOptions.Value.Password.RequiredLength` to 6, then the generated password will not pass identity validation. For that reason, I thought I would just find the largest value between the two. In most cases the above line is equal to `IdentityOptions.Value.Password.RequiredLength`. I just thought I would add a bullet proof code even covering a very rare/strange case.

Reply:
I assume this is a developer issue, all will expect that `IdentityOptions.Value.Password.RequiredLength` is the length that has been used. So, I can't image someone will set the length to 9 and required unique character 11 unless he 🚬 something ;)

Reply:
okay. I'll changed it and wait until that smoken developer submit an issue :)